extension/show-names.js

```diff
@@ -0,0 +1,57 @@
+window.showRealNames = () => {
+	const storageKey = 'cachedNames';
+
+	const getCachedUsers = cb => {
+		chrome.storage.local.get(storageKey, data => cb(data[storageKey]));
+	};
+
+	const updateCachedUsers = users => {
+		chrome.storage.local.set({[storageKey]: users});
+	};
+
+	const addUsersName = (user, name) => {
+		const usernameLinks = document.querySelectorAll(`.timeline-comment-header-text:not(.has-full-name) a[href="/${user}"]`);
+		for (const userLink of usernameLinks) {
+			const nameSpan = `<span class="comment-full-name">(${name}) -</span>`;
+			userLink.insertAdjacentHTML('afterend', nameSpan);
+			userLink.closest('.timeline-comment-header-text').classList.add('has-full-name');
+		}
+	};
+
+	getCachedUsers(users => {
+		const usersOnPage = Array.from(document.querySelectorAll('.js-discussion .author')).map(el => el.innerText);
+		const uniqueUsers = new Set(usersOnPage);
+
+		// Add cached users to DOM first, since the calls for everyone else will be slow
+		for (const user of uniqueUsers) {
+			const cachedName = users[user];
+			if (cachedName) {
+				addUsersName(user, cachedName);
+				uniqueUsers.delete(user);
+			}
+		}
+
+		const userUrl = user => `//github.com/${user}`;
+		const requests = Array.from(uniqueUsers).map(user => fetch(userUrl(user)).then(res => res.text()));
+		Promise.all(requests).then(profiles => {
+			const userCache = {};
+
+			for (const profile of profiles) {
+				const frag = document.createRange().createContextualFragment(profile);
+				const fullnameEl = frag.querySelector('.vcard-fullname');
+				const usernameEl = frag.querySelector('.vcard-username');
+				const fullname = fullnameEl && fullnameEl.innerText;
+				const username = usernameEl && usernameEl.innerText;
+
+				// Possible to not have name el in response if user comments under an org account.
```

Comment:
I should make that comment more clear. I guess what I meant to say was, for users commenting as an `org`, the `vcard-fullname` class actually doesn't exist in the returned markup at all, so `innerText` would throw an exception. Exception shouldn't be an issue once I switch this over to Sprint.